node/src/main/scala/org/bitcoins/node/messages/data/HeadersMessage.scala

```diff
@@ -15,16 +15,18 @@ import scodec.bits.ByteVector
 object HeadersMessage extends Factory[HeadersMessage] {
   private case class HeadersMessageImpl(
       count: CompactSizeUInt,
-      headers: List[BlockHeader])
+      headers: Vector[BlockHeader])
```

Comment:
We've discussed this before, but I still think its a good idea to not be as picky as to what data structure we expect. This change causes a ripple effect to other places in the code, and although it may be a good thing in this specific case it might not always be that way. Wouldn't it be better to take in a `Seq`/`Iterable` in places like this? We could continue being picky about what data structures we prefer at _call site_, without sacrificing API flexibility.